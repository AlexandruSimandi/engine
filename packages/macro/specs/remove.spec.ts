import pluginTester from "babel-plugin-tester";
import prettier from "prettier";
import plugin from "babel-plugin-macros";

const macroFile = "'@c11/engine.macro'";

pluginTester({
  plugin,
  babelOptions: { filename: __filename },
  formatResult: (result: any) => {
    return prettier.format(result, {
      parser: "babel",
    });
  },
  tests: {
    "should support Get": {
      code: `
        import { producer } from ${macroFile}
        const result = producer((
          a1 = Remove.foo
        ) => { })
      `,
      snapshot: true,
    },
  },
});
