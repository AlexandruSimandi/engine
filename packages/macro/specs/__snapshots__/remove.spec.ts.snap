// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros should support Get: should support Get 1`] = `

import { producer } from '@c11/engine.macro'
const result = producer((
  a1 = Remove.foo
) => { })

      ↓ ↓ ↓ ↓ ↓ ↓

const result = {
  args: {
    type: "STRUCT",
    value: {
      a1: {
        type: "REMOVE",
        path: [
          {
            type: "CONST",
            value: "foo"
          }
        ]
      }
    },
    meta: {
      order: ["a1"]
    }
  },
  fn: a1 => {}
};


`;
